doc/mgmt/SONiC-Management-Framework-Image-mgmt-HLD.md

```diff
@@ -0,0 +1,250 @@
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+**image install**
+
+This command is used to install a new image on the alternate image partition. This command takes a path to an installable SONiC image or URL and installs the image.
+
+**show image-list**
+
+This command displays information about currently installed images. It displays a list of installed images, currently running image and image set to be loaded in next reboot.
+
+**image set-default**
+
+This command is be used to change the image which can be loaded by default in all the subsequent reboots.
+
+**image set-next-boot**
+
+This command is used to change the image that can be loaded in the *next* reboot only. Note that it will fallback to current image in all other subsequent reboots after the next reboot.
+
+**image remove**
+
+This command is used to remove the unused SONiC image from the disk. Note that it's *not* allowed to remove currently running image.
+
+Note: The above text is taken from the sonic command line reference Wiki (https://github.com/Azure/sonic-utilities/blob/master/doc/Command-Reference.md#sonic-installer)
+
+## 2.2 Functional Description
+After recieving the request from the client, via an RPC, the rest server will transfer the control to processAction method in the app module(inside traslib). This method will parse the target uri path and will branch to the corresponding function. These functions will call the python scripts in the host to perform image management related actions, like install, remove ..etc. The response from the output of the script is propagated back to processAction method and is converted to json. The json message is sent back to the client.
+
+# 3 Design
+## 3.1 Overview
+TBD
+
+## 3.2 DB Changes
+
+N/A
+### 3.2.1 CONFIG DB
+
+N/A
+
+### 3.2.2 APP DB
+
+N/A
+
+### 3.2.3 STATE DB
+
+N/A
+
+### 3.2.4 ASIC DB
+
+N/A
+
+### 3.2.5 COUNTER DB
+
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+
+N/A
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+```
+ +---x image-install
+    |  +---w input
+    |  |  +---w installer-arg-type?   string
+    |  |  +---w filename?             string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x image-remove
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x image-default
+    |  +---w input
+    |  |  +---w imagename?   string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x image-nextboot
+    |  +---w input
+    |  |  +---w imagename?   string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x image-list
+       +--ro output
+          +--ro images-found?   string
+```
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
```

Comment:
Can a command to boot to ONIE prompt on next reboot be added?

Reply:
Sure, we will try to add it but after buzznik